### ISSUE TITLE
[BUG] Middleware - Experimental Branch Example gives response error on reload

### ISSUE THREAD

**Issue by himanshu-ntml** [opennextjs/opennextjs-cloudflare#155]
### Describe the bug

I encountered an issue while running the middleware example from the experimental branch. On every reload of the page, the response throws an error. I've attached a video recording to help illustrate the problem.

https://github.com/user-attachments/assets/d317b7b6-a0e3-48aa-aa82-5f6786c7a041

### Steps to reproduce

Steps to Reproduce:

Clone the repository and switch to the experimental branch.
Run the middleware example using pnpm run preview.
Open the browser and navigate to the application.
Refresh the page multiple times.

### Expected behavior

The middleware should handle requests and provide a consistent response without any errors.

### @opennextjs/cloudflare version

experimental

### Node.js version

v22.11.0

### Wrangler version

3.93.0

### next info output

```shell
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.1.0: Thu Oct 10 21:03:15 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T6000
  Available memory (MB): 65536
  Available CPU cores: 10
Binaries:
  Node: 22.11.0
  npm: 10.9.0
  Yarn: 1.22.22
  pnpm: 9.12.3
Relevant Packages:
  next: 14.2.11 // An outdated version detected (latest is 15.0.4), upgrade is highly recommended!
  eslint-config-next: 14.2.14
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.5.4
Next.js Config:
  output: N/A
 ⚠ An outdated version detected (latest is 15.0.4), upgrade is highly recommended!
   Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
   Read more - https://nextjs.org/docs/messages/opening-an-issue
```

**Comment by vicb** [comment]
Can you try with cache disabled? 
I think I mentioned the known issue in the readme.

**Comment by himanshu-ntml** [comment]
Thank you for the quick reply! Apologies for missing the notes earlier. I’m just curious—how does that impact the response? Is there any solution to address this? My main goal in experimenting with the branch is to test ISR, so I’ll make sure to try it out for that as well.

**Comment by himanshu-ntml** [comment]
@vicb is there is any way to replicate disable cache behaviour using code? Thank you so much

**Comment by vicb** [comment]
This was fixed in main / 0.3 release via https://github.com/opennextjs/opennextjs-cloudflare/pull/195
